src/features/movies/MovieList/index.js

```diff
@@ -1,55 +1,77 @@
 import { useEffect } from "react";
 import { useDispatch, useSelector } from "react-redux";
-import { fetchMovies, selectError, selectGenres, selectLoading, selectMovies } from "./popularMoviesSlice";
+import {
+  fetchMovies,
+  selectError,
+  selectLoading,
+  selectGenres,
+  selectMovies,
+  selectTotalResults,
+} from "./popularMoviesSlice";
 import { Section } from "../../../common/Section";
 import { MovieWrapper } from "../../../common/MovieWrapper";
 import { Title } from "../../../common/Title";
 import { Loader } from "../../../common/Loader";
 import { ErrorPage } from "../../../common/ErrorPage";
 import { MovieTile } from "../../../common/MovieTile";
 import { APIImageUrl } from "../../APIdata";
+import searchQueryParamName from "../../../common/Search/searchQueryParamName";
+import { NoResultsPage } from "../../../common/NoResultsPage";
+import { useQueryParameter } from "../../../common/Search/queryParameterHooks";
 import { nanoid } from "@reduxjs/toolkit";
 
 export const MovieList = () => {
   const dispatch = useDispatch();
-
+  const query = useQueryParameter(searchQueryParamName);
+  
   useEffect(() => {
-    dispatch(fetchMovies());
-  }, [dispatch])
-
-  const popularMovies = useSelector(selectMovies);
+    dispatch(fetchMovies({ query }));
+  }, [dispatch, query]);
+  
   const genresTable = useSelector(selectGenres);
   const loading = useSelector(selectLoading);
   const error = useSelector(selectError);
+  const movies = useSelector(selectMovies);
+  const totalResults = useSelector(selectTotalResults);
 
   return (
-    <Section>
-      {error
-        ? <ErrorPage />
-        : <>
-          <Title>Popular movies</Title>
-          {loading
-            ? <Loader />
-            : <MovieWrapper>
-              {
-                popularMovies.map((movie) => (
-                  <MovieTile
-                    key={nanoid()}
-                    poster={`${APIImageUrl}/w342${movie.poster_path}`}
-                    posterPath={movie.poster_path}
-                    title={movie.title}
-                    date={movie.release_date.slice(0, 4)}
-                    voteAverage={movie.vote_average}
-                    voteCount={`${movie.vote_count} votes`}
-                    genres={genresTable.filter((genre) => movie.genre_ids.includes(genre.id))}
-                    id={movie.id}
-                  />
-                ))
-              }
-            </MovieWrapper>
-          }
-        </>
-      }
-    </Section>
+      <Section>
+        {query && error || totalResults === 0
```

Comment:
You fixed a bug in displaying the ResultPage when it shouldn't, but now error handling doesn't work :(.

Reply:
We don't want NoResultPage to be displayed when there is an error - we want the ErrorPage